tdi_python/tdiTable.py

```diff
@@ -0,0 +1,2617 @@
+                        else:
+                            to_print += "\n    {}{} : {}".format(indent, name.ljust(30), f_val)
+            return to_print
+
+        def stringify_output(self, value):
+            if isinstance(value, str):
+                return value
+            if self.category == "data" and (self.table.data_type_map(self.data_type) in ["INT_ARR", "BOOL_ARR"] or
+                                           (self.table.data_type_map(self.data_type) == "BYTE_STREAM" and ('$bfrt_field_class', 'register_data') in self.annotations)):
+                return self._stringify_int_arr(value)
+            elif ((self.category == "key" and self.table.key_type_map(self.type) == "EXACT") or (self.category == "data" and self.table.data_type_map(self.data_type) in ["UINT64", "BYTE_STREAM"])):
+                return self._stringify_int(value)
+            elif self.category == "key" and self.table.key_type_map(self.type) in ["TERNARY", "RANGE", "LPM"]:
+                p0 = self._stringify_int(value[0])
+                p1 = self._stringify_int(value[1])
+                return p0, p1
+            elif self.category == "key" and self.table.key_type_map(self.type) in ["OPTIONAL"]:
+                p0 = self._stringify_int(value[0])
+                p1 = self._stringify_bool(value[1])
+                return p0, p1
+            elif self.category == "data" and self.table.data_type_map(self.data_type) == "FLOAT":
+                return self._stringify_float(value)
+            elif self.category == "data" and self.table.data_type_map(self.data_type) == "BOOL":
+                return self._stringify_bool(value)
+            elif self.category == "data" and self.table.data_type_map(self.data_type) == "STRING":
+                return str(value)
+            elif self.category == "data" and self.table.data_type_map(self.data_type) == "STR_ARR":
+                return self._stringify_str_arr(value)
+            elif self.category == "data" and self.table.data_type_map(self.data_type) == "CONTAINER":
+                return self._stringify_cont(value, self.name)
+            else: # don't raise errors in case user is working with multiple entries
+                if self.category == "data":
+                    print("Error: type {} for data field {} not yet supported in the CLI.".format(self.table.data_type_map(self.data_type), self.name))
+                elif self.category == "key":
+                    print("Error: type {} for key field {} not yet supported in the CLI.".format(self.table.key_type_map(self.type), self.name))
+                else:
+                    print("Error: unknown field type - category={}".format(self.category))
+                return "Error"
+
+
+    """
+    These maps are based on the enums defined in tdi_table.h If the
+    enums there are changed, these maps must also be changed.
+    """
+    @staticmethod
+    def key_type_map(key_type):
+        if key_type == 0:
+            return "EXACT"
```

Comment:
Let's think of a better way to create these maps rather than have them as hardcoded. Maybe we can keep enum to string descriptor mappings in the tdi_info_parser itself for Match type, data type and API type.
Similar mapping for table type and flag type in target specific code but a common API in the TDI interface

Reply:
Can keep as python dictionary map in this way be fast and can be extended to platform specific json file defined externally?
Currently I just defined as dict.